karapace/avro_dataclasses/introspect.py

```diff
@@ -115,13 +114,14 @@ def _field_type(field: Field, type_: object) -> AvroType:  # pylint: disable=too
             raise UnderspecifiedAnnotation("Key and value types must be specified for map types")
         if args[0] is not str:
             raise UnsupportedAnnotation("Key type must be str")
-        return MapType(
-            {
-                "type": "map",
-                "values": _field_type(field, args[1]),
-                **({"default": field.default_factory()} if field.default_factory is not MISSING else {}),
-            }
-        )
+        map_dict: MapType = {
+            "name": field.name,
```

Comment:
This was previously missing and somehow didn't fail, despite it being required. `field.name` is an educated guess, please double check.